services/autorust/codegen/src/codegen_operations.rs

```diff
@@ -664,90 +625,29 @@ fn create_operation_code(cg: &CodeGen, operation: &WebOperationGen) -> Result<Op
             autorust_openapi::StatusCode::Default => {}
         }
     }
-    for (status_code, rsp) in &error_responses {
-        match status_code {
-            autorust_openapi::StatusCode::Code(_) => {
-                let tp = create_response_type(rsp)?.map(TypeNameCode::into_token_stream);
-                let rsp_value = create_rsp_value(tp.as_ref());
-                let status_code_name = get_status_code_ident(status_code)?;
-                let response_type_name = get_response_type_ident(status_code)?;
-                match tp {
-                    Some(_tp) => {
-                        match_status.extend(quote! {
-                            http::StatusCode::#status_code_name => {
-                                let rsp_body = azure_core::collect_pinned_stream(rsp_stream).await.map_err(Error::ResponseBytes)?;
-                                #rsp_value
-                                Err(Error::#response_type_name{value: rsp_value})
-                            }
-                        });
-                    }
-                    None => {
-                        match_status.extend(quote! {
-                            http::StatusCode::#status_code_name => {
-                                Err(Error::#response_type_name{})
-                            }
-                        });
-                    }
-                }
-            }
-            autorust_openapi::StatusCode::Default => {}
+    match_status.extend(quote! {
+        status_code => {
+            Err(azure_core::error::Error::from(azure_core::error::ErrorKind::HttpResponse { status: status_code.as_u16(), error_code: None }))
```

Comment:
It would be nice to have a `.into_error()` for an `ErrorKind`. I'm punting on `error_code` here to a future PR. I think core needs to slights adjustments.